libcontainer/cgroups/fs/devices.go

```diff
@@ -26,7 +26,11 @@ func (s *DevicesGroup) Apply(d *cgroupData) error {
 }
 
 func (s *DevicesGroup) Set(path string, cgroup *configs.Cgroup) error {
-	if system.RunningInUserNS() {
+	pids, err := cgroups.GetPids(path)
+	if err != nil {
+		return err
+	}
+	if system.RunningInUserNS(pids[0]) {
```

Comment:
@cyphar 
Thanks for the comments.
I will correct the code, but configs.Cgroup itself part of cgroupData as per struct definition. 
pid field is not part of configs.Cgroup
type cgroupData struct {
        root      string
        innerPath string
        config    *configs.Cgroup
        pid       int
}
Without affecting set interface, I'll look for better option to get the pid of running container